extra/resources/ifspeed

```diff
@@ -165,13 +182,40 @@ monitor() {
     return ${ret}
 }
 
+# This function tries to guess nic interface by IP
+get_nic_name_by_ip(){
+    # $FINDIF takes its parameters from the environment
+    NICINFO=`${FINDIF}`
+    rc=$?
+    if [ $rc -eq 0 ];then
+        NICINFO=`echo "${NICINFO}" | sed -e 's/netmask\ //;s/broadcast\ //'`
+        echo "${NICINFO}" | cut -d" " -f1
+    else
+        echo ""
+    fi
+}
+
 validate() {
     # Check the interface parameter
-    if [ -z "${OCF_RESKEY_iface}" ]; then 
-        ocf_log err "Empty iface parameter.  Please specify network interface to check"
-        exit $OCF_ERR_CONFIGURED
+    if [ -z "${OCF_RESKEY_iface}" ]; then
+        if [ -z "${OCF_RESKEY_ip}" ]; then
+            ocf_log err "Empty iface and ip parameters. Please specify either an interface name or valid ip address."
+            exit $OCF_ERR_CONFIGURED
+        else
+            # Try to guess interface from IP
+            nic=`get_nic_name_by_ip`
+            if [ -z "${nic}" ];then
+                # findif couldn't find the interface
+                if ocf_is_probe; then
+                    ocf_log info "[$FINDIF] failed"
+                    exit $OCF_NOT_RUNNING
+                else
+                    ocf_log err "[$FINDIF] failed"
+                    exit $rc
+                fi
+            fi
+        fi
```

Comment:
validate() should just check that the configuration is sane, i.e. that either iface or ip is specified. The rest can go into update()

One benefit is that stops won't fail if the IP is gone